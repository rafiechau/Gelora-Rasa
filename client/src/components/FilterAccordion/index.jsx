import React from 'react';
import PropTypes from 'prop-types';
import {
  Accordion,
  AccordionDetails,
  AccordionSummary,
  RadioGroup,
  FormControlLabel,
  Radio,
  Typography,
} from '@mui/material';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';

const FilterAccordion = ({ title, options, selectedOption, onChange }) => (
  <Accordion
    data-testid="filter-accordion"
    sx={{ backgroundColor: '#333', color: 'white', borderRadius: '8px', marginBottom: 2 }}
  >
    <AccordionSummary expandIcon={<ExpandMoreIcon style={{ color: 'white' }} />} data-testid="accordion-summary">
      <Typography>{title}</Typography>
    </AccordionSummary>
    <AccordionDetails>
      <RadioGroup value={selectedOption} onChange={onChange} data-testid="radio-group">
        {options.map((option) => (
          <FormControlLabel
            key={option.id}
            value={option.value}
            control={<Radio />}
            label={option.label}
            style={{ marginRight: '12px' }}
            data-testid={`radio-option-${option.id}`}
          />
        ))}
      </RadioGroup>
    </AccordionDetails>
  </Accordion>
);

FilterAccordion.propTypes = {
  title: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
      value: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
  selectedOption: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default FilterAccordion;
