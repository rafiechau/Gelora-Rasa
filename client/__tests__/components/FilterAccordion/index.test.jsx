// FilterAccordion.test.jsx
import { render, screen, fireEvent } from '@testing-library/react';
import FilterAccordion from '@components/FilterAccordion';

describe('FilterAccordion', () => {
  const mockOptions = [
    { id: '1', value: 'option1', label: 'Option 1' },
    { id: '2', value: 'option2', label: 'Option 2' },
  ];
  const mockOnChange = jest.fn();

  beforeEach(() => {
    render(
      <FilterAccordion title="Test Title" options={mockOptions} selectedOption="option1" onChange={mockOnChange} />
    );
  });

  test('should render accordion with given title', () => {
    expect(screen.getByTestId('filter-accordion')).toBeInTheDocument();
    expect(screen.getByText('Test Title')).toBeInTheDocument();
  });

  test('should have correct number of radio options', () => {
    const radioOptions = screen.getAllByTestId((id) => id.startsWith('radio-option-'));
    expect(radioOptions).toHaveLength(mockOptions.length);
  });

  test('should call onChange when a radio option is selected', () => {
    const secondOption = screen.getByTestId('radio-option-2');
    fireEvent.click(secondOption);
    expect(mockOnChange).toHaveBeenCalled();
  });

  test('should match snapshot', () => {
    const { asFragment } = render(
      <FilterAccordion title="Test Title" options={mockOptions} selectedOption="option1" onChange={mockOnChange} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
