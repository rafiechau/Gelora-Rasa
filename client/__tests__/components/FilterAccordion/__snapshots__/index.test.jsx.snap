// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterAccordion should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-mmkaam-MuiPaper-root-MuiAccordion-root"
    data-testid="filter-accordion"
  >
    <div
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
      data-testid="accordion-summary"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Test Title
        </p>
      </div>
      <div
        class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          style="color: white;"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="MuiAccordion-region"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
            >
              <div
                class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
                data-testid="radio-group"
                role="radiogroup"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  data-testid="radio-option-1"
                  style="margin-right: 12px;"
                >
                  <span
                    class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary Mui-checked MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      name=":r4:"
                      type="radio"
                      value="option1"
                    />
                    <span
                      class="css-hyxlzm"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                        data-testid="RadioButtonUncheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11zohuh-MuiSvgIcon-root"
                        data-testid="RadioButtonCheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Option 1
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  data-testid="radio-option-2"
                  style="margin-right: 12px;"
                >
                  <span
                    class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-1m9pwf3"
                      name=":r4:"
                      type="radio"
                      value="option2"
                    />
                    <span
                      class="css-hyxlzm"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                        data-testid="RadioButtonUncheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                        data-testid="RadioButtonCheckedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    Option 2
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
